zeroex/orderwatch/order_watcher.go

```diff
@@ -56,30 +82,75 @@ type Watcher struct {
 	orderValidator             *ordervalidator.OrderValidator
 	wasStartedOnce             bool
 	mu                         sync.Mutex
+	maxExpirationTime          *big.Int
+	maxExpirationCounter       *slowcounter.SlowCounter
+	maxOrders                  int
+}
+
+type Config struct {
+	MeshDB            *meshdb.MeshDB
+	BlockWatcher      *blockwatch.Watcher
+	OrderValidator    *ordervalidator.OrderValidator
+	NetworkID         int
+	ExpirationBuffer  time.Duration
+	MaxOrders         int
+	MaxExpirationTime *big.Int
 }
 
 // New instantiates a new order watcher
-func New(meshDB *meshdb.MeshDB, blockWatcher *blockwatch.Watcher, orderValidator *ordervalidator.OrderValidator, networkID int, expirationBuffer time.Duration) (*Watcher, error) {
+func New(config Config) (*Watcher, error) {
 	decoder, err := decoder.New()
 	if err != nil {
 		return nil, err
 	}
 	assetDataDecoder := zeroex.NewAssetDataDecoder()
-	contractAddresses, err := ethereum.GetContractAddressesForNetworkID(networkID)
+	contractAddresses, err := ethereum.GetContractAddressesForNetworkID(config.NetworkID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Apply defaults to config.
+	if config.MaxOrders == 0 {
+		config.MaxOrders = defaultMaxOrders
+	}
+	if config.MaxExpirationTime == nil {
+		config.MaxExpirationTime = constants.UnlimitedExpirationTime
+	} else if big.NewInt(time.Now().Unix()).Cmp(config.MaxExpirationTime) == 1 {
+		// MaxExpirationTime should never be in the past.
+		config.MaxExpirationTime = big.NewInt(time.Now().Unix())
```

Comment:
`Unix` already converts the time to UTC. Since it's unclear how we are going to treat expiration buffers moving forward I would rather leave this as is for now.

Reply:
I was talking about the existing buffer config but realize that it doesn't make sense to add it here.